README.md

```diff
@@ -30,77 +30,96 @@ export FAROS_CLI_VERSION="v0.0.1"
-| --start_time \<start>                  | FAROS_START_TIME                | Now                         | all        |
-| --end_time \<end>                      | FAROS_END_TIME                  | Now                         | all        |
-| --app_platform \<platform>             | FAROS_APP_PLATFORM              | "NA"                        | all        |
-| --deployment \<deployment>             | FAROS_DEPLOYMENT                | Random UUID                 | deployment |
-| --deployment_env_details \<details>    | FAROS_DEPLOYMENT_ENV_DETAILS    | ""                          | deployment |
-| --deployment_status_details \<details> | FAROS_DEPLOYMENT_STATUS_DETAILS | ""                          | deployment |
-| --deployment_start_time \<start>       | FAROS_DEPLOYMENT_START_TIME     | FAROS_START_TIME            | deployment |
-| --deployment_end_time \<end>           | FAROS_DEPLOYMENT_END_TIME       | FAROS_END_TIME              | deployment |
-| --build \<build>                       | FAROS_BUILD                     | FAROS_COMMIT_SHA            | build      |
-| --build_status_details \<details>      | FAROS_BUILD_STATUS_DETAILS      | ""                          | build      |
-| --build_start_time \<start>            | FAROS_BUILD_START_TIME          | FAROS_START_TIME            | build      |
-| --build_end_time \<end>                | FAROS_BUILD_END_TIME            | FAROS_END_TIME              | build      |
-
-#### :wrench: Additional Settings Flags
-
-| Flag          | Description                            |
-| ------------- | -------------------------------------- |
-| --dry_run     | Print the event instead of sending.    |
-| -s / --silent | Unexceptional output will be silenced. |
-| --debug       | Helpful information will be printed.   |
-
-### Usage Examples
-
-#### :mega: Sending a build event
-
-##### Using flags
+### :book: Event Types
+
+An event type (i.e. `deployment`) corresponds to the step of your CI/CD process that you are instrumenting. Each event type represents a set of fields (required and optional) that are used to populate a specific set of Faros' canonical models which are then sent to Faros. The event type is the main argument passed to the cli. Below are the supported event types with their required and optional fields.
+
+> :exclamation: Important: Every event type requires the "General Required Fields" and can optionally set the "General Optional Fields".
+
+There are two ways that fields can be passed into the script. The first, is via environment variables. The second is via flags. You may use a combination of these two options. If both are set, flags will take precedence over environment variables.
+
+By convention, you can switch between using a flag or an environment variable by simple capitalizing the argument name and prefixing it with FAROS_. For example, --commit_sha becomes FAROS_COMMIT_SHA, --vcs_org becomes FAROS_VCS_ORG.
+
+> General Required Fields
```

Comment:
1. Let just call them `Required Fields`.
2. Switch to `####`, i.e `#### Required Fields`

Same for `General Optional Fields` -> `Optional Fields`